### PR TITLE
Support new media structure

### PR DESCRIPTION
**Pull request by RemcoSimonides**
- [x] Migrate media on CMS
- [x] New media form

### CODE REVIEW

File: apps/cms/src/app/template/template.component.ts [hunk 1]
```diff
@@ -78,6 +79,33 @@ export class TemplateComponent implements OnInit, OnDestroy {
 
   save() {
     const params = this.route.snapshot.params as CmsParams;
-    this.service.save(this.form.value, params);
+    const value = this.updateMediaMetadata(this.form.value, params)
+    this.service.save(value, params);
+  }
+
+  updateMediaMetadata(value: CmsTemplate<Section>, params: CmsParams) {
```
Comment (RemcoSimonides) [inline apps/cms/src/app/template/template.component.ts:86]: Only when saving we know which section has media. This is when we insert the correct index in `field`.

**Review by pldespaigne — APPROVED** [review]

File: apps/cms/src/app/template/template.component.ts [hunk 2]
```diff
@@ -78,6 +79,33 @@ export class TemplateComponent implements OnInit, OnDestroy {
...
+    const { app, page, template } = params;
+    const getCollection = () => ['cms', app, page].filter(v => !!v).join('/');
```
Comment (GrandSchtroumpf) [inline apps/cms/src/app/template/template.component.ts:88]: if you don't put any params, you don't need to create a function for that. The value of the collection won't change depending on the section : 
```typescript
const collection = ['cms', app, page].filter(v => !!v).join('/');
```

File: apps/cms/src/app/template/template.component.ts [hunk 3]
```diff
@@ -78,6 +79,33 @@ export class TemplateComponent implements OnInit, OnDestroy {
...
+
+    value.sections.forEach((section, i) => {
+      let fields = [];
```
Comment (GrandSchtroumpf) [inline apps/cms/src/app/template/template.component.ts:91]: Better to have a mapping for that : 
```typescript
// Outside of the component
const mediaFields = {
  banner: ['background', 'image']
  hero: ['background']
}
....
const fields = mediaFields[section._type] ?? [];
```

File: apps/cms/src/app/template/template.component.ts [hunk 4]
```diff
@@ -78,6 +79,33 @@ export class TemplateComponent implements OnInit, OnDestroy {
...
+      if (section._type === 'banner') {
+        fields = ['background', 'image'];
+      } else if (section._type === 'hero') {
+        fields = ['background'];
+      }
+
+      fields.forEach(field => {
```
Comment (GrandSchtroumpf) [inline apps/cms/src/app/template/template.component.ts:98]: Prefere `for (const field of fields)` over forEach if you don't need the index.

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**